### DISCUSSION TITLE
Generate Classes using Roslyn

### DISCUSSION THREAD

**Discussion by poetbyron** [CollaboratingPlatypus/PetaPoco#659] [Ideas]
I understand that T4 templates went away in V6.  However, this cumbersome task with large databases is extremely valuable.  Could the code be migrated to generate the POCOs using Roslyn as explained in the article below:

https://carlos.mendible.com/2017/03/02/create-a-class-with-net-core-and-roslyn/

**Comment by pleb** [comment]
@poetbyron Did you have a look at [PetaPoco.DBEntityGenerator](https://github.com/hippasus/PetaPoco.DBEntityGenerator) in the [Add ons](https://github.com/CollaboratingPlatypus/PetaPoco#add-ons) section?

**Reply by poetbyron** [reply]
That's exactly what I'm looking for. Thank you!
